File: src/main/java/nl/shadowlink/tools/shadowlib/paths/Nodes.java
package nl.shadowlink.tools.shadowlib.paths;

import nl.shadowlink.tools.io.ReadFunctions;

/**
 * @author Shadow-Link
 */
public class Nodes {
    private int nodeCount;
    private int carNodeCount;
    private int intersectionNodeCount;
    private int linkCount;

    public Nodes() {
        ReadFunctions rf = new ReadFunctions("E:/Nodes/nodes29.nod");

        readHeader(rf);
        for (int i = 0; i < nodeCount; i++) {
            Node node = new Node(rf);
        }
        for (int i = 0; i < linkCount; i++) {
            Link link = new Link(rf);
        }
    }

    private void readHeader(ReadFunctions rf) {
        nodeCount = rf.readInt();
        carNodeCount = rf.readInt();
        intersectionNodeCount = rf.readInt();
        linkCount = rf.readInt();
        System.out.println("Node Count: " + nodeCount);
        System.out.println("Car Node Count: " + carNodeCount);
        System.out.println("intersectionNode Count: " + intersectionNodeCount);
        System.out.println("Link Count: " + linkCount);
    }

}

File: src/main/java/nl/shadowlink/tools/shadowlib/paths/Nodes.kt
package nl.shadowlink.tools.shadowlib.paths

import nl.shadowlink.tools.io.ReadFunctions

/**
 * @author Shadow-Link
 */
class Nodes {
    private var nodeCount = 0
    private var carNodeCount = 0
    private var intersectionNodeCount = 0
    private var linkCount = 0

    init {
        val rf = ReadFunctions("E:/Nodes/nodes29.nod")
        readHeader(rf)
        for (i in 0 until nodeCount) {
            val node = Node(rf)
        }
        for (i in 0 until linkCount) {
            val link = Link(rf)
        }
    }

    private fun readHeader(rf: ReadFunctions) {
        nodeCount = rf.readInt()
        carNodeCount = rf.readInt()
        intersectionNodeCount = rf.readInt()
        linkCount = rf.readInt()
        println("Node Count: $nodeCount")
        println("Car Node Count: $carNodeCount")
        println("intersectionNode Count: $intersectionNodeCount")
        println("Link Count: $linkCount")
    }
}